src/System.Net.Primitives/src/System/Net/CookieCollection.cs

```diff
@@ -97,7 +86,7 @@ public void Add(CookieCollection cookies)
             {
                 throw new ArgumentNullException("cookies");
             }
-            foreach (Cookie cookie in cookies)
+            foreach (Cookie cookie in cookies._list)
```

Comment:
Note that someone could pass in a subclass of `CookieCollection` here that provides its own `GetEnumerator()`, in which case we'd no longer be calling the subclass's `GetEnumerator()`.

We could protect against this with something like...

```c#
if (cookies.GetType() == typeof(CookieCollection))
{
    foreach (Cookie cookie in cookies._list)
    {
        Add(cookie);
    }
}
else
{
    foreach (Cookie cookie in cookies)
    {
        Add(cookie);
    }
}
```

But I suspect that nobody would ever do something like this so why even bother having the extra code to supporting it.